library/test/src/lib.rs

```diff
@@ -506,7 +506,24 @@ pub fn run_test(
         let supports_threads = !cfg!(target_os = "emscripten") && !cfg!(target_arch = "wasm32");
         if concurrency == Concurrent::Yes && supports_threads {
             let cfg = thread::Builder::new().name(name.as_slice().to_owned());
-            Some(cfg.spawn(runtest).unwrap())
+            let mut runtest = Arc::new(Mutex::new(Some(runtest)));
+            let runtest2 = runtest.clone();
+            cfg.spawn(move || runtest2.lock().unwrap().take().unwrap()())
+                .map_or_else(
+                    |e| {
+                        if e.kind() == io::ErrorKind::WouldBlock {
+                            // `ErrorKind::WouldBlock` means hitting the thread limit on some
+                            // platforms, so run the test synchronously here instead.
+                            Arc::get_mut(&mut runtest).unwrap().get_mut().unwrap().take().unwrap()(
+                            );
```

Comment:
This is done by rustfmt.

Reply:
Wow, that's really bad formatting. Maybe you can add `#[rustfmt::skip]` to that line or perhaps add a line comment (and then a newline) after `Arc::get_mut(...)` to get rustfmt to break the line?

Reply:
I opened a rustfmt issue for this: 
https://github.com/rust-lang/rustfmt/issues/4678